### DISCUSSION TITLE
Doesn't Kratos work for multi sub domain ?

### DISCUSSION THREAD

**Discussion by meotimdihia** [ory/kratos#1327] [Q&A] [answered]
There is the guide to configure for multi-domain cookies: 
https://www.ory.sh/kratos/docs/guides/multi-domain-cookies

But if Kratos is running on subdomains, then it doesn't work. For example:
https://domain1.com => https://kratos.domain1.com
https://domain2.com => https://kratos.domain2.com

While set cookies can't do it with multi-domain:

> WARNING
> Do not set the cookie domain when using multiple domain aliases as it enforces the domain - breaking sessions on your alias domains.

Do we have a way to configure this?  my websites are using serverless: `next.js`

**Comment by aeneasr** [comment, marked as answer]
Please share your config

**Reply by meotimdihia** [reply]
@aeneasr: Hi, I upload it to here: https://github.com/meotimdihia/test-ory-kratos

The example works only if I add the session configs to `kratos.yaml`

```
session:
  cookie:
    domain: loca.lt
```

I used 3 domains: 
https://meotimdihia.loca.it -> example node run on port 4455
https://meotimdihia4433.loca.it -> public API
https://meotimdihia4434.loca.it -> admin API

**Reply by aeneasr** [reply]
I see, maybe the docs are confusing there. So what you need to do is this: https://www.ory.sh/kratos/docs/guides/multi-domain-cookies/#cookies

With domain aliasing, we are referencing this: https://www.ory.sh/kratos/docs/guides/multi-domain-cookies/#base-url

Domain aliasing is only useful when you have multiple TLDs - so `foo.com` `bar.com`. If you have on common domain `loca.it` you don't need it

**Reply by meotimdihia** [reply]
@aeneasr: that is my examples repository but I have 2 domains with the same user database in the production. 
The cookies config doesn't work in my case cause it doesn't support multi-domain.

**Reply by aeneasr** [reply]
Are they hosted on two individual TLDs? If so, can you please share the two TLDs?

**Reply by meotimdihia** [reply]
My 2 domains are two individual TLDs.
They run on Vercel serverless but I don't implement Kratos with them yet.
I am evaluating Kratos for replacing existing authentication.

**Reply by aeneasr** [reply]
Can you share the TLD patterns?

**Reply by meotimdihia** [reply]
> Can you share the TLD patterns?

Hi, 2 domains are: 

`https://anime**.fun` and  `https://comic**.fun`
I want to run Kratos on `https://auth.anime**.fun` and `https://auth.comic**.fun` that shared users database

**Reply by aeneasr** [reply]
Ok, do you want users also to be authenticated at `comic` if they log in to `anime`? If so, that's currently not possible because cookies are isolated per TLD. We have an issue for that open: https://github.com/ory/kratos/issues/662

**Reply by meotimdihia** [reply]
@aeneasr I just want to use the same user database. I don't need that users log in once and to be authenticated on both domains.
I don't know if we have a workaround for this?

**Reply by Benehiko** [reply]
@meotimdihia I think you should be able to run two kratos instances connected to the same database. This would share the same users across your two domains.
